scripts/utils/Timer.js

```diff
@@ -1,94 +1,88 @@
-
+// See details: https://medium.com/sablier/writing-accurate-time-dependent-truffle-tests-8febc827acb5
 class Timer {
   constructor(web3) {
       this.web3 = web3;
   }
 }
 
-Timer.prototype.takeSnapshot = function() {
-  return new Promise( (resolve, reject) => {
-    this.web3.currentProvider.send({
-      jsonrpc: '2.0',
-      method: 'evm_snapshot',
-      id: new Date().getTime()
-    }, (err, snapshotId) => {
-      if (err) { return reject(err) }
-      return resolve(snapshotId)
-    })
-  });
-}
-
-Timer.prototype.advanceTime = function(time) {
-  return new Promise( (resolve, reject) => {
-    this.web3.currentProvider.send({
-      jsonrpc: '2.0',
-      method: 'evm_increaseTime',
-      params: [time],
-      id: new Date().getTime()
-    }, (err, result) => {
-      if (err) { return reject(err) }
-      return resolve(result)
-    })
-  })
-}
+const handleResultOrError = async (web3, title, {resolve, reject}, err, result, log = (_) => {}) => {
+  if (err || result.error || !result.result) {
+    console.log(`${title} - Error: ${JSON.stringify(result)}`);
+    return reject(err);
+  }
+  const block = await web3.eth.getBlock("latest");
+  console.log(`Latest block #: ${block.number}`);
+  log(block);
+  return resolve(result);
+};
 
-Timer.prototype.advanceBlock = function() {
+Timer.prototype.takeSnapshot = function () {
+  console.log(`Taking blockchain snapshot.`);
   return new Promise((resolve, reject) => {
-    this.web3.currentProvider.send({
-      jsonrpc: '2.0',
-      method: 'evm_mine',
-      id: new Date().getTime()
-    }, (err, result) => {
-      if (err) { return reject(err) }
-      return resolve(result)
-    })
+    this.web3.currentProvider.send(
+      {
+        jsonrpc: "2.0",
+        method: "evm_snapshot",
+        id: new Date().getTime(),
+      },
+      async (err, snapshotId) => {
+        await handleResultOrError(
+          this.web3,
+          'TakingSnapshot',
+          { resolve, reject },
+          err,
+          snapshotId,
+          () => console.log(`Taking blockchain snapshot ID ${snapshotId.result}. Result: ${JSON.stringify(snapshotId)}`)
+        );
+      },
+    );
   });
-}
+};
 
-Timer.prototype.advanceBlockAndSetTime = function(time) {
+Timer.prototype.advanceBlockAtTime = function (time) {
```

Comment:
`advanceBlockAtTime` isnt an intuitive name to me... what does this function do? Advance the block time to the provided time?